include/util/enum.hpp

```diff
@@ -1,26 +1,25 @@
 #pragma once
 
+#include <cctype>
 #include <iostream>
 #include <map>
+#include <stdexcept>
 #include <string>
 
 namespace waybar::util {
 
+template <typename EnumType>
 struct EnumParser {
   EnumParser() {}
 
-  enum SORT_METHOD { ID, NAME, NUMBER, DEFAULT };
-
-  SORT_METHOD sortStringToEnum(const std::string& str) {
-    // Convert the input string to uppercase (make it lenient on config input)
+  EnumType sortStringToEnum(const std::string& str,
+                            const std::map<std::string, EnumType>& enumMap) {
+    // Convert the input string to uppercase
     std::string uppercaseStr;
     for (char c : str) {
-        uppercaseStr += std::toupper(c);
+      uppercaseStr += std::toupper(c);
     }
 
-    static const std::map<std::string, SORT_METHOD> enumMap = {
-        {"ID", ID}, {"NAME", NAME}, {"NUMBER", NUMBER}, {"DEFAULT", DEFAULT}};
-
     auto it = enumMap.find(uppercaseStr);
     if (it != enumMap.end()) {
       return it->second;
```

Comment:
on the next line, you can remove the `else {` because you're returning here.

Reply:
and then you can also remove the {} of the else

Reply:
not sure I understand the ask here. The if else is just for determining if they passed in a valid setting and whether to throw an invalid argument exception to bubble up to the caller to determine how to handle

Reply:
nvm... you mean if it never hits the return it will throw the error.